src/main/java/seedu/weeblingo/logic/commands/EndCommand.java

```diff
@@ -19,9 +19,13 @@
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
-        requireNonNull(model);
-        model.updateFilteredFlashcardList(PREDICATE_SHOW_ALL_FLASHCARDS);
-        return new CommandResult(MESSAGE_SUCCESS, false, false, false);
+        if (model.getMode().getCurrentMode() == 2 || model.getMode().getCurrentMode() == 3) {
```

Comment:
not too important but probably can import the static final variable MODE_QUIZ and MODE_LEARN so that the code looks nicer? :D same for the rest of the magic numbers